### PR TITLE
Refactor Option::filter method

### PR DESCRIPTION
**Pull request by shanavas786**

### CODE REVIEW

File: src/libcore/option.rs [hunk 1]
```diff
@@ -634,16 +634,12 @@ impl<T> Option<T> {
...
-                } else {
-                    None
-                }
+        if let Some(x) = self {
+            if predicate(&x) {
+                return Some(x)
             }
-            None => None,
-        }
+        };
```
Comment (sinkuu) [inline src/libcore/option.rs]: Nitpicking: `;` is unnecessary here.

Comment (shanavas786) [reply src/libcore/option.rs]: Yeah, Updated. Thanks

**Review by sugar700 — APPROVED** [review]

**Comment by LukasKalbertodt** [comment]
What was the reason for this change? I don't really care that my code was changed, I just don't really understand why. I *guess* it's because it has fewer lines and only one `None`? Was this change discussed in IRC? I would *really* appreciate it, if PRs for such changes would have at least a short description. Otherwise the rest of the world has no clue what's going on.

**Comment by bstrie** [comment]
@LukasKalbertodt I surmise it was just mildly more idiomatic to use `if let` to reduce the slight amount of repetition.

**Comment by Kerollmops** [comment]
Why not this thought ?

```rust
match self {
    Some(x) if predicate(&x) => Some(x),
    _ => None
}
```

**Comment by sinkuu** [comment]
@Kerollmops That would be cetainly better, but currently impossible (produces an error E0008).

**Comment by Kerollmops** [comment]
Ho ! you're right ! sorry ! I hope the NLL will change this kind of error.

**Pull request merged**